src/coreclr/jit/fginline.cpp

```diff
@@ -1568,13 +1581,179 @@ void Compiler::fgInsertInlineeBlocks(InlineInfo* pInlineInfo)
         //
         bottomBlock->RemoveFlags(BBF_DONT_REMOVE);
 
+        // If the inlinee has EH, merge the EH tables, and figure out how much of
+        // a shift we need to make in the inlinee blocks EH indicies.
+        //
+        unsigned const inlineeRegionCount = InlineeCompiler->compHndBBtabCount;
+        const bool     inlineeHasEH       = inlineeRegionCount > 0;
+        unsigned       inlineeIndexShift  = 0;
+
+        if (inlineeHasEH)
+        {
+            // If the call site also has EH, we need to insert the inlinee clauses
+            // so they are a child of the call site's innermost enclosing region.
+            // Figure out what this is.
+            //
+            bool           inTryRegion     = false;
+            unsigned const enclosingRegion = ehGetMostNestedRegionIndex(iciBlock, &inTryRegion);
+
+            // We will insert the inlinee clauses in bulk before this index.
+            //
+            unsigned insertBeforeIndex = 0;
+
+            if (enclosingRegion == 0)
+            {
+                // The call site is not in an EH region, so we can put the inlinee EH clauses
+                // at the end of root method's the EH table.
+                //
+                // For example, if the root method already has EH#0, and the inlinee has 2 regions
+                //
+                //   enclosingRegion   will be 0
+                //   inlineeIndexShift will be 1
+                //   insertBeforeIndex will be 1
+                //
+                //   inlinee eh0 -> eh1
+                //   inlinee eh1 -> eh2
+                //
+                //   root eh0 -> eh0
+                //
```

Comment:
Is this true, and if so, desirable? What if the root method has:
```
try { } ... // #1
call()
try { } ... // #2
```
and `call` has EH and is inlined. Don't we want to put the `call` EH info in between the EH info for `#1` and `#2` in the resulting EH table? So, the lexical order is respected?

Or maybe the lexical order doesn't matter for EH regions without nesting relationships, since we already presumably can move them around ("make them out of order") today.

Reply:
It's cheaper to add entries at the end of the table. A mid-table insert requires walking the flow graph to update EH indices and remapping all the ACDs (though there are none this early).

What we really could use is a persistent notion of EH region identity, but I've so far avoided thinking about EH redesigns.